lib/lune.js

```diff
@@ -80,6 +80,10 @@ function tokm (au) {
 function phase (date) {
   if (!date) {
     date = new Date()
+  } else if (!(date instanceof Date)) {
+    throw new Error('Invalid parameter');
+  } else if (date.toString() === 'Invalid Date') {
+    throw new Error('Invalid Date');
```

Comment:
For whatever it's worth, I think a better way to do this would be as follows:

```javascript
if (!date) {
  date = new Date()
}
if(!(date instanceof Date)) {
  throw new TypeError("Invalid parameter")
}
if(Number.isNaN(date.getTime())) {
  throw new RangeError("Invalid Date");
}
```

This is for three reasons:
1. These are not things that should be checked if/else... we should verify that they all pass!
2. It is pretty slow (comparatively speaking) to cast the date to a string for validation. Better to just if the time that the Date object represents is valid.
3. Normally `Error` is for generic runtime errors. We should try to be more specific, and I recommend `TypeError` (you passed us an argument of the wrong type) and `RangeError` (you passed us an argument of the right type but out of valid range).